pkg/bpf/bpffs.go

```diff
@@ -72,11 +72,37 @@ func MapPrefixPath() string {
 	return filepath.Join(mapRoot, mapPrefix)
 }
 
+// MapPath returns a path for a BPF map with a given name.
+// NOTE: This function should be used only by cilium daemon.
 func MapPath(name string) string {
 	once.Do(lockDown)
 	return filepath.Join(mapRoot, mapPrefix, name)
 }
 
+// MapPathCLI returns a path for a BPF map with a given name.
+// NOTE: This function should be used by CLI or any component which is not
+// daemon.
+func MapPathCLI(name string) (string, error) {
```

Comment:
Instead of doing this, can we detect inside of `MapPath()` whether we run as part of cilium-agent and fall back to using mountinfo instead? I really dislike encoding CLI logic into the generic bpf package.

If we are not running in the agent and we can't rely on mountinfo we might as well just error out with Fatal from within the `MapPath()` function.